src/block/block_compact.c

```diff
@@ -205,9 +205,25 @@ __block_compact_skip_internal(WT_SESSION_IMPL *session, WT_BLOCK *block, bool es
     __wt_verbose_level(session, WT_VERB_COMPACT,
       (estimate ? WT_VERBOSE_DEBUG_3 : WT_VERBOSE_DEBUG_1),
       "%s:%s require 10%% or %" PRIuMAX "MB (%" PRIuMAX
-      ") in the first 90%% of the file to perform compaction, compaction %s",
+      ") in the first 90%% of the file to perform compaction",
       block->name, estimate ? " estimating --" : "", (uintmax_t)(file_size / 10) / WT_MEGABYTE,
-      (uintmax_t)(file_size / 10), *skipp ? "skipped" : "proceeding");
+      (uintmax_t)(file_size / 10));
+
+    /*
+     * Skip files that have failed to make progress on previous compact iterations.
+     */
+    if (!estimate && block->compact_estimated && !*skipp) {
```

Comment:
I don't think we need to check the two bool variables:
```suggestion
    if (block->compact_estimated && !*skipp) {
```